src/bluesky/tests/test_run_engine.py

```diff
@@ -1956,6 +1957,45 @@ def plan():
             RE(plan())
 
 
+def test_wait_with_planstub(RE):
+    def runtime_error_and_success_plan():
+        async def passing_coro():
+            await asyncio.sleep(0.001)
+            return 1080
+
+        passing_task = asyncio.create_task(passing_coro())
+
+        async def failing_coro():
+            await asyncio.sleep(0.001)
+            raise RuntimeError
+
+        failing_task = asyncio.create_task(failing_coro())
+
+        tasks = yield from wait_for([lambda: passing_task, lambda: failing_task])
+        assert passing_task is tasks[0] and failing_task is tasks[1]
+        assert passing_task.done() and passing_task.result() == 1080
+        assert failing_task.done() and isinstance(failing_task.exception(), RuntimeError)
```

Comment:
Now you've made the returning futures change, please can we check that passing coros and returning classes works:
```suggestion
    def runtime_error_and_success_plan():
        async def passing_coro():
            await asyncio.sleep(0.001)
            return 1080

        async def failing_coro():
            await asyncio.sleep(0.001)
            raise RuntimeError

        passing_task, failing_task = yield from wait_for([lambda: passing_coro(), lambda: failing_coro()])
        assert passing_task.get_coro().__name__ == "passing_coro"
        assert passing_task.done() and passing_task.result() == 1080
        assert failing_task.get_coro().__name__ == "failing_coro"
        assert failing_task.done() and isinstance(failing_task.exception(), RuntimeError)
```